pkg/cri/server/container_remove.go

```diff
@@ -48,6 +48,12 @@ func (c *criService) RemoveContainer(ctx context.Context, r *runtime.RemoveConta
 		return nil, fmt.Errorf("get container info: %w", err)
 	}
 
+	sandboxID := container.SandboxID
+	sandbox, err := c.sandboxStore.Get(sandboxID)
+	if err != nil {
+		return nil, fmt.Errorf("can't find sandbox for sandbox ID %s: %w", sandboxID, err)
```

Comment:
IMO we need to be able to remove the container anyway... not seeing a good reason to abort the removal of a stopped container if the containers pod is no longer in the store.

I'm good with a log entry here but now aborting the removal or stop

Will kubelet be ok if we send a container deleted without a sandbox id or 0/nil?

Reply:
Currently kubelet will panic with nil pointer if podSandboxStatus is nil: https://github.com/kubernetes/kubernetes/blob/86284d42f891cac285c08c8a350ca8724f12c21f/pkg/kubelet/pleg/evented.go#L207

Sync'ed with @mikebrow over slack. It is possible that a podSandbox cannot be found when cri tries to stop/remove a container. This could happen when 
1. a pod sandbox (s1) is being created but not yet finished,
2. a container (c1) is being created under the pod sandbox s1.
3. s1 is deleted. Note that since step 2 has not finished yet, it is possible that RemovePodSandbox did not remove the c1.
4. c1 is created. 
5. kubelet requests to delete c1. Since now s1 is deleted, containerd cannot find the sandbox for `c1.SandboxID`. 

Agree that we should not change the behavior here (i.e. we should just log an error and continue removing, if sandbox cannot be found.) I will change the code, and send a PR to kubernetes to make kubelet tolerant of this behavior.

Reply:
Done updating the code. Keeping the comment unresolved until I create the kubernetes PR to make evented PLEG tolerate `podSandboxStatuses == nil`